### PR TITLE
Re-fetch statuses on opening data entry home page

### LINKED ISSUE #336: Finalise button ("Invoerfase afronden") only appears after reload
After completing the data entry for the last polling station, the finalise button ("Invoerfase afronden") is not shown right away. It only appears after a reload of the polling station choice page.

Same is happening with the status badges of the polling stations (see #347).

### PR DESCRIPTION
**Pull request by praseodym**
Fixes #336

### CODE REVIEW

**Comment by jschuurk-kr** [comment]
When I go to `/overview` and click on one of the elections, the call to get the status of the election I selected (e.g. `/elections/2/status`) is made twice.

**Comment by praseodym** [comment]
> When I go to `/overview` and click on one of the elections, the call to get the status of the election I selected (e.g. `/elections/2/status`) is made twice.

Indeed, but unfortunately it's not that easy to fix. The `DataEntryHomePage` component only gets rendered once the initial statuses request completes, and then immediately calls `refetch()` to update the statuses with the second statuses request. 

If we don't block the `DataEntryHomePage` render we can let that do the initial load and prevent a second request, but it will cause the footer to be already rendered before the request completes. I think that's not trivial to fix: it would require something like [`<Suspense>`](https://react.dev/reference/react/Suspense) which is only usable with a query framework (such as [TanStack Query](https://tanstack.com/query/latest/docs/framework/react/guides/suspense)) because there is no official API yet (see ['Note' block in Suspense docs](https://react.dev/reference/react/Suspense#displaying-a-fallback-while-content-is-loading)).

**Comment by jschuurk-kr** [comment]
> > When I go to `/overview` and click on one of the elections, the call to get the status of the election I selected (e.g. `/elections/2/status`) is made twice.
> 
> Indeed, but unfortunately it's not that easy to fix. [\<snip\>](https://github.com/kiesraad/abacus/pull/349#issuecomment-2357769113)

Fixing it sounds worse than leaving as it is. So works for me to keep the two calls.

**Review by jschuurk-kr — DISMISSED** [review]

**Comment by Lionqueen94** [comment]
I see I forgot to rename renderInputHomePage, @praseodym can you do this for me? 😇

**Comment by praseodym** [comment]
> I see I forgot to rename renderInputHomePage, @praseodym can you do this for me? 😇

Done: be66bb13d58cb97eca6f953dbc87753cd51a9d73

**Review by jschuurk-kr — APPROVED** [review]

**Review by Lionqueen94 — APPROVED** [review]

File: frontend/lib/api/election/ElectionStatusProvider.tsx [hunk 1]
```diff
@@ -14,20 +15,20 @@ export interface ElectionStatusProviderProps {
...
   });
 
-  if (loading) {
+  if (data === null) {
     return null;
   }
 
-  if (!data || error) {
-    throw new Error();
+  if (error) {
```
Comment (lkleuver) [inline frontend/lib/api/election/ElectionStatusProvider.tsx:26]: shouldn't this be defined above `if (data === null) return null;` ? (just in case)

**Review by lkleuver — APPROVED** [review]

File: frontend/lib/api/election/ElectionStatusProvider.tsx (hunk 1 above)
Comment (praseodym) [reply frontend/lib/api/election/ElectionStatusProvider.tsx:26]: I'll create a PR to fix that!

Comment (praseodym) [reply frontend/lib/api/election/ElectionStatusProvider.tsx:26]: #354

**Pull request merged**